commit bb68922f100e78982515e7593de9fd23b58172d8
Author: Sean Paul <seanpaul@chromium.org>
Date:   Tue Jan 30 09:29:34 2018 -0500

    drm/i915: Restore HDCP DRM_INFO when with no downstream
    
    The commit below returned earlier than before, but failed to move the
    info message when authenticating without downstream devices. This patch
    restores the message on authentication success.
    
    Changes in v2:
    - s/no downstream devices/no repeater present/ (Ram)
    
    Fixes: 87eb3ec818fa ("drm/i915: II stage HDCP auth for repeater only")
    Cc: Ramalingam C <ramalingam.c@intel.com>
    Cc: Sean Paul <seanpaul@chromium.org>
    Cc: Jani Nikula <jani.nikula@linux.intel.com>
    Cc: Joonas Lahtinen <joonas.lahtinen@linux.intel.com>
    Cc: Rodrigo Vivi <rodrigo.vivi@intel.com>
    Cc: intel-gfx@lists.freedesktop.org
    Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
    Signed-off-by: Sean Paul <seanpaul@chromium.org>
    Link: https://patchwork.freedesktop.org/patch/msgid/20180130142943.168314-1-seanpaul@chromium.org
    Link: https://patchwork.freedesktop.org/patch/msgid/20180130142943.168314-1-seanpaul@chromium.org

diff --git a/drivers/gpu/drm/i915/intel_hdcp.c b/drivers/gpu/drm/i915/intel_hdcp.c
index db9527173a1e..9fc9da030514 100644
--- a/drivers/gpu/drm/i915/intel_hdcp.c
+++ b/drivers/gpu/drm/i915/intel_hdcp.c
@@ -168,10 +168,8 @@ int intel_hdcp_auth_downstream(struct intel_digital_port *intel_dig_port,
 
 	/* If there are no downstream devices, we're all done. */
 	num_downstream = DRM_HDCP_NUM_DOWNSTREAM(bstatus[0]);
-	if (num_downstream == 0) {
-		DRM_INFO("HDCP is enabled (no downstream devices)\n");
+	if (num_downstream == 0)
 		return 0;
-	}
 
 	ksv_fifo = kzalloc(num_downstream * DRM_HDCP_KSV_LEN, GFP_KERNEL);
 	if (!ksv_fifo)
@@ -502,6 +500,7 @@ static int intel_hdcp_auth(struct intel_digital_port *intel_dig_port,
 	if (repeater_present)
 		return intel_hdcp_auth_downstream(intel_dig_port, shim);
 
+	DRM_INFO("HDCP is enabled (no repeater present)\n");
 	return 0;
 }